File: play-2.5.x/K-Modules/kmodules/K-Base/app/K/Controllers/PlanTask/PlanTaskNotify.java
package K.Controllers.PlanTask;

import K.Controllers.JsonpController;
import K.Service.PlanTasks.PlanTaskService;
import K.Service.ServiceMgr;
import play.mvc.Result;

/**
 * Created by kk on 15/12/2.
 */
public class PlanTaskNotify extends JsonpController{

    public Result NewPlanTask() {
        PlanTaskService planTaskService = (PlanTaskService) ServiceMgr.Instance.getService(PlanTaskService.ServiceName);
        if (planTaskService != null) {
            planTaskService.NotifyNewTask();
        }
        return ok("OK");
    }

}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Controllers/PlanTask/PlanTaskNotify.kt
package K.Controllers.PlanTask

import K.Controllers.JsonpController
import K.Service.PlanTasks.PlanTaskService
import K.Service.ServiceMgr
import play.mvc.Result
import play.mvc.Results

/**
 * Created by kk on 15/12/2.
 */
class PlanTaskNotify : JsonpController() {

    fun NewPlanTask(): Result {
        val planTaskService = ServiceMgr.Instance.getService(PlanTaskService.ServiceName) as PlanTaskService
        planTaskService?.NotifyNewTask()
        return Results.ok("OK")
    }

}
